drivers/include/net/netdev2_ieee802154.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup drivers_netdev_netdev2
+ * @brief
+ * @{
+ *
+ * @file
+ * @brief   Definitions for netdev2 common IEEE 802.15.4 code
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NETDEV2_IEEE802154_H_
+#define NETDEV2_IEEE802154_H_
+
+#include "net/ieee802154.h"
+#include "net/gnrc/nettype.h"
+#include "net/netopt.h"
+#include "net/netdev2.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    IEEE 802.15.4 netdev2 flags
+ * @brief   Flags for netdev_ieee802154_t::flags
+ *
+ * The flag-space `0xff00` is available for device-specific flags.
+ * The flag-space `0x00ff` was chosen for global flags to be in accordance to
+ * the IEEE 802.15.4 MAC header flags.
+ * @{
+ */
+
+#define NETDEV2_IEEE802154_SEND_MASK        (0x0068)    /**< flags to take for send packets */
+#define NETDEV2_IEEE802154_RESV1            (0x0001)    /**< reserved flag */
+#define NETDEV2_IEEE802154_RAW              (0x0002)    /**< pass raw frame to upper layer */
+/**
+ * @brief   use long source addres (set) or short source address (unset)
+ */
+#define NETDEV2_IEEE802154_SRC_MODE_LONG    (0x0004)
+#define NETDEV2_IEEE802154_SECURITY_EN      (0x0008)    /**< enable security */
+#define NETDEV2_IEEE802154_RESV2            (0x0010)    /**< reserved flag */
+
+/**
+ * @brief   request ACK from receiver
+ */
+#define NETDEV2_IEEE802154_ACK_REQ          (0x0020)
+#define NETDEV2_IEEE802154_PAN_COMP         (0x0040)    /**< compress source PAN ID */
+#define NETDEV2_IEEE802154_RESV3            (0x0080)    /**< reserved flag */
+/**
+ * @}
+ */
+
+/**
+ * @brief Extended structure to hold IEEE 802.15.4 driver state
+ *
+ * @extends netdev2_t
```

Comment:
Why not? Because whenever I want to change `netdev2_t` now I have also to change all strucs that inherit directly or indirectly from this struct.
Why I didn't say anything? Because I wasn't involved in the design.